src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -341,6 +398,40 @@ internal static bool TryCreate(SyntheticBoundNodeFactory F, MethodSymbol method,
             return true;
         }
 
+        private static MethodSymbol GetCustomCreateMethod(
+            SyntheticBoundNodeFactory F,
+            NamedTypeSymbol builderType)
+        {
+            // The Create method's return type is expected to be buildderType.
```

Comment:
Fixed in https://github.com/dotnet/roslyn/pull/13405/commits/eef0c70b2966698aa71be1063a920290e5778f51#diff-995d1bf33206eef8c980dbd08b8601b0L405